### PR TITLE
quilt ssh should fail for partially booted machines

### PR DESCRIPTION
**Pull request by tejasmanohar**
Resolves [quilt ssh should fail for partially booted machines (Trello)](https://trello.com/c/zzPrg9MR/314-quilt-ssh-should-fail-for-partially-booted-machines).

### CODE REVIEW

File: cluster/provider/cloud_config.go [hunk 1]
```diff
@@ -112,14 +112,14 @@ date >> /var/log/bootscript.log
 
 export DEBIAN_FRONTEND=noninteractive
 
+ssh_keys="%[2]s"
+setup_user quilt "$ssh_keys"
```
Comment (kklin) [inline cluster/provider/cloud_config.go]: Doesn't `setup_user` add the user to the docker group, which doesn't exist until docker is installed?

Comment (tejasmanohar) [inline cluster/provider/cloud_config.go]: Dropped that commit since it's no longer necessary with 47250d8

Comment (tejasmanohar) [inline cluster/provider/cloud_config.go]: actually may be good anyways so ssh is available asap, will correct

File: cluster/foreman.go [hunk 2]
```diff
@@ -108,9 +109,17 @@ func (fm *foreman) runOnce() {
...
+
 		if connected && !m.connected {
+			// Check that SSH service has started.
+			addr := fmt.Sprintf("%s:ssh", m.machine.PublicIP)
+			conn, err := net.Dial("tcp", addr)
+			if err != nil {
+				log.WithError(err).Debugf("No service running at %s", addr)
+				return
+			}
+			defer conn.Close()
```
Comment (kklin) [inline cluster/foreman.go]: No need to defer

File: cluster/provider/cloud_config.go (hunk 1 above)
Comment (tejasmanohar) [inline cluster/provider/cloud_config.go]: fixed

**Comment by tejasmanohar** [comment]
@kklin Addressed concerns.

**Comment by tejasmanohar** [comment]
ah, missed the tests on this fn. will re-open when ready

**Pull request closed without merging**